tests/externalsystems/OfferProvider.Library.Tests/BusinessLogic/OfferProviderBusinessLogicTests.cs

```diff
@@ -217,49 +217,64 @@ public async Task TriggerProviderCallback_WithNoServiceAccountSet_CallsExpected(
         result.nextStepTypeIds.Should().BeNull();
         result.stepStatusId.Should().Be(ProcessStepStatusId.DONE);
         result.modified.Should().BeTrue();
-        A.CallTo(() => _offerProviderService.TriggerOfferProviderCallback(A<OfferProviderCallbackData>.That.Matches(x => x.TechnicalUserInfo == null), A<string>._, A<CancellationToken>._))
+        A.CallTo(() => _offerProviderService.TriggerOfferProviderCallback(A<OfferProviderCallbackData>.That.Matches(x => x.TechnicalUsersInfo == null), A<string>._, A<CancellationToken>._))
             .MustHaveHappenedOnceExactly();
 
     }
 
     [Fact]
-    public async Task TriggerProviderCallback_WithMultipleServiceAccountSet_Throws()
+    public async Task TriggerProviderCallback_WithValidData_InternalSA_ReturnsExpected()
     {
         // Arrange
-        var fakeId = Guid.NewGuid();
+        var technicalUserId = Guid.NewGuid();
+        var technicalUserClientId = "sa1";
+        var technicalUserInternalClientId = Guid.NewGuid().ToString();
         var serviceAccounts = new (Guid, string?, TechnicalUserKindId)[]
         {
-            new(Guid.NewGuid(), "sa1", TechnicalUserKindId.INTERNAL),
-            new(Guid.NewGuid(), "sa2", TechnicalUserKindId.INTERNAL)
+            new(technicalUserId, technicalUserClientId, TechnicalUserKindId.INTERNAL)
         };
-        A.CallTo(() => _offerSubscriptionRepository.GetTriggerProviderCallbackInformation(fakeId))
+        A.CallTo(() => _offerSubscriptionRepository.GetTriggerProviderCallbackInformation(_subscriptionId))
             .Returns((serviceAccounts, "cl1", "https://callback.com", OfferSubscriptionStatusId.ACTIVE));
-        async Task Act() => await _sut.TriggerProviderCallback(fakeId, CancellationToken.None);
+        A.CallTo(() => _provisioningManager.GetIdOfCentralClientAsync(technicalUserClientId))
+            .Returns(technicalUserInternalClientId);
+        A.CallTo(() => _provisioningManager.GetCentralClientAuthDataAsync(technicalUserInternalClientId))
+            .Returns(new ClientAuthData(IamClientAuthMethod.SECRET) { Secret = "test123" });
 
         // Act
-        var ex = await Assert.ThrowsAsync<ConflictException>(Act);
+        var result = await _sut.TriggerProviderCallback(_subscriptionId, CancellationToken.None);
 
         // Assert
-        ex.Message.Should().Be("There should be not more than one service account for the offer subscription");
+        result.nextStepTypeIds.Should().BeNull();
+        result.stepStatusId.Should().Be(ProcessStepStatusId.DONE);
+        result.modified.Should().BeTrue();
+        A.CallTo(() => _offerProviderService.TriggerOfferProviderCallback(A<OfferProviderCallbackData>.That.Matches(x => x.TechnicalUsersInfo!.FirstOrDefault()!.TechnicalUserSecret == "test123"), A<string>._, A<CancellationToken>._))
```

Comment:
inappropriate use of !
```suggestion
                A.CallTo(() => _offerProviderService.TriggerOfferProviderCallback(A<OfferProviderCallbackData>.That.Matches(x => x.TechnicalUsersInfo != null && x.TechnicalUsersInfo.Count() > 0 && x.TechnicalUsersInfo.First().TechnicalUserSecret == "test123"), A<string>._, A<CancellationToken>._))
```